src/services/contract-service.js

```diff
@@ -276,8 +277,15 @@ class ContractService {
       method
         .send(opts)
         .on('receipt', resolve)
-        .on('confirmation', confirmationCallback)
-        .on('transactionHash', transactionHashCallback)
+        //.on('confirmation', confirmationCallback)
+        //.on('transactionHash', transactionHashCallback)
+        // Workaround for "confirmationCallback" not being triggered in some versions of Metamask
+        .on('transactionHash', (hash) => {
+          if (transactionHashCallback)
+            transactionHashCallback(hash)
+          if (confirmationCallback)
+            this.checkForDeploymentCompletion(hash, confirmationCallback)
```

Comment:
nit: since now it is not only for deploy, can we rename checkForDeploymentCompletion to something more generic such as confirmationWatcher or something equivalent that does not have "deployment" in its name ?

Reply:
Yup thanks will do.

Reply:
also to nit: 
- Could we have the name including "polling", since that's the distinguishing feature vs the callback method?
- On 284, 286: We've been avoiding if clauses without `{}` brackets, and I think `npm run format` enforces this. 

And just to be clear, this fix _will_ use the callback method for older versions of metamask, but will fall back to polling if this isn't possible? Where exactly is this decision made in the code?